### ISSUE TITLE
Asynchronously read and write files at the same time, write failures in juicefs cluster

### ISSUE THREAD

**Issue by zml1206** [juicedata/juicefs#3737]
**What happened**:
In juicefs cluster, when spark writes a file, another thread continues to read the file at the same time, 
The result of spark running is that the task is successful, but the file occasional fail to be write, and there is no record of deleting the file in the .trash.
It's okay to read after writing.
**What you expected to happen**:
At the beginning, because the file has not been generated, the reading failed. After the file is generated, it will be read successfully.

**How to reproduce it (as minimally and precisely as possible)**:
First, there is a resident spark application, which requests to perform write operations through http, and reads asynchronously at the same time.
Shell emulation:
```
#!/bin/bash
curl --request POST 'http://localhost:8090/jobs?sync=true' --header 'Content-Type: application/json' --data '{"pool": "default","jobId": "637361e0-fdfe-11ed-9a13-3f5e76c379bb", "script": "spark.range(1, 10).write.mode(\"overwrite\").parquet(\"/juicefs/test-bucket/tmp/t1\")"}' &
while true
do
cat /juicefs/test-bucket/tmp/t1/*.parquet
done
```

**Anything else we need to know?**

**Environment**:
- JuiceFS version: 1.0.4+2023-04-06.f1c475d
- metadata engine: mysql
- mysql isolation level: REPEATABLE-READ
- JuiceFS mode: cluster
- storage: minio

**Comment by SandyXSD** [comment]
It's possible to run `echo` and `mv` first, and then `rm` to delete the t2.txt. Do not use pipe here, instead use `;` or `&&` to execute commands sequentially.

**Comment by davies** [comment]
In HDFS, the rename should NOT overwrite the existing file, which is different from POSIX, so you have to delete the existing files before rename, or it will fail.

**Comment by zml1206** [comment]
> It's possible to run `echo` and `mv` first, and then `rm` to delete the t2.txt. Do not use pipe here, instead use `;` or `&&` to execute commands sequentially.

There is a problem with my reproduce, I will describe it again.

**Comment by zml1206** [comment]
I have changed the steps to reproduce. @davies @SandyXSD

**Comment by zml1206** [comment]
There is no error when spark writes, but the file does not exist, nor does it exist in metadata or Trash

**Comment by tangyoupeng** [comment]
spark.range(1, 10).write.mode(\"overwrite\").parquet(\"/juicefs/test-bucket/tmp/t1\")"}
This will write data to defautfs

**Comment by tangyoupeng** [comment]
What's the defaultfs for the spark?

**Comment by zml1206** [comment]
Defaultfs is /juicefs, It's okay when to read after writing.

**Comment by tangyoupeng** [comment]
Are you in our wechat group?

**Comment by zml1206** [comment]
How to join?

**Comment by tangyoupeng** [comment]
https://go.juicefs.com/slack

**Comment by SandyXSD** [comment]
Fixed by #3864
